e2e/pushpull_test.go

```diff
@@ -87,6 +95,103 @@ func runWithDindSwarmAndRegistry(t *testing.T, todo func(dindSwarmAndRegistryInf
 
 }
 
+func TestPushArchs(t *testing.T) {
+	runWithDindSwarmAndRegistry(t, func(info dindSwarmAndRegistryInfo) {
+		testCases := []struct {
+			name              string
+			args              []string
+			expectedPlatforms []manifestlist.PlatformSpec
+		}{
+			{
+				name: "default",
+				args: []string{},
+				expectedPlatforms: []manifestlist.PlatformSpec{
+					{
+						OS:           "linux",
+						Architecture: "amd64",
+					},
+				},
+			},
+			{
+				name: "all-platforms",
+				args: []string{"--all-platforms"},
+				expectedPlatforms: []manifestlist.PlatformSpec{
+					{
+						OS:           "linux",
+						Architecture: "amd64",
+					},
+					{
+						OS:           "linux",
+						Architecture: "386",
+					},
+					{
+						OS:           "linux",
+						Architecture: "ppc64le",
+					},
+					{
+						OS:           "linux",
+						Architecture: "s390x",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v5",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v6",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm",
+						Variant:      "v7",
+					},
+					{
+						OS:           "linux",
+						Architecture: "arm64",
+						Variant:      "v8",
+					},
+				},
+			},
+		}
+
+		for _, testCase := range testCases {
+			t.Run(testCase.name, func(t *testing.T) {
+				cmd := info.configuredCmd
+				ref := info.registryAddress + "/test/push-pull:1"
+				args := []string{"app", "push", "--tag", ref, "--insecure-registries=" + info.registryAddress}
+				args = append(args, testCase.args...)
+				args = append(args, filepath.Join("testdata", "push-pull", "push-pull.dockerapp"))
+				cmd.Command = dockerCli.Command(args...)
+				icmd.RunCmd(cmd).Assert(t, icmd.Success)
+
+				http := &http.Client{}
+				r, err := http.Get("http://" + info.registryAddress + "/v2/test/push-pull/manifests/1")
+				assert.NilError(t, err)
```

Comment:
Could you split this into 2 parts ? One fetching/checking the bundle manifest and the other fetching/checking the manifest list platforms?